apps/src/maze/harvester.js

```diff
@@ -0,0 +1,200 @@
+import Gatherer from './gatherer';
+import HarvesterCell from './harvesterCell';
+import HarvesterDrawer from './harvesterDrawer';
+import mazeMsg from './locale';
+import { HarvesterTerminationValue } from '../constants.js';
+import { TestResults } from '../constants.js';
+
+export default class Harvester extends Gatherer {
+
+  /**
+   * @override
+   */
+  getCellClass() {
+    return HarvesterCell;
+  }
+
+  /**
+   * @override
+   */
+  createGridItemDrawer() {
+    return new HarvesterDrawer(this.maze_.map, this.skin_, this);
+  }
+
+  hasCorn(id) {
+    return this.hasHarvest(HarvesterCell.FeatureType.CORN, id);
+  }
+
+  hasPumpkin(id) {
+    return this.hasHarvest(HarvesterCell.FeatureType.PUMPKIN, id);
+  }
+
+  hasBean(id) {
+    return this.hasHarvest(HarvesterCell.FeatureType.BEAN, id);
+  }
+
+  hasHarvest(crop, id) {
```

Comment:
nit: It seems like this should be named `hasCrop` (and below we would have `atCrop`, `gotCropAt` and `getCrop`).